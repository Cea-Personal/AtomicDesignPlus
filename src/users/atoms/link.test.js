import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Linked from "./link";
afterEach(cleanup);
function renderWithRouter(ui) {
  return {
    ...render(<Router>{ui}</Router>)
  };
}
it("display a link", () => {
  const { getByText, asFragment, container } = renderWithRouter(
    <Linked value={"Sign In"} link={"sign_in"} />
  );
  expect(container.innerHTML).toMatch("Sign In");
  fireEvent.click(getByText(/sign in/i));
  expect(asFragment()).toMatchSnapshot();
});
