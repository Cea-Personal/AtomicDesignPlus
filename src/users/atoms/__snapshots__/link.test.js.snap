// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display a link 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa gqQkip"
  >
    <a
      href="/sign_in"
    >
      Sign In
    </a>
  </div>
</DocumentFragment>
`;
